### PR TITLE
Fix attention head display issues

### PR DESCRIPTION
**Pull request by ddaspit**
- compute width of tokens instead of using fixed char width
- input and output tokens were swapped when computing link positions

### CODE REVIEW

**Comment by ddaspit** [comment]
@googlebot I signed it!

**Comment by iftenney** [comment]
Thanks for sending this! Adding @EmilyReif to take a look

**Comment by iftenney** [comment]
Possible fix for CJK display issue; see #28

File: lit_nlp/client/modules/attention_module.ts [hunk 1]
```diff
@@ -204,6 +204,16 @@ export class AttentionModule extends LitModule {
     // clang-format on
   }
 
+  private getTokOffsets(tokWidths: number[], spaceWidth: number): number[] {
```
Comment (EmilyReif) [inline lit_nlp/client/modules/attention_module.ts:207]: Add to utils.ts? Might be useful for other visualizations

**Review by EmilyReif — APPROVED** [review]
Looks great! Thanks so much for doing this.

File: lit_nlp/client/lib/utils.ts [hunk 2]
```diff
@@ -310,3 +310,14 @@ export function linkifyUrls(
   ret.push(text.slice(lastIndex, text.length));
   return html`${ret}`;
 }
+
+const CANVAS = document.createElement('canvas');
```
[the remark this answers is not in the document]
Comment (ddaspit) [reply lit_nlp/client/lib/utils.ts:314]: Done

File: lit_nlp/client/modules/attention_module.ts (hunk 1 above)
Comment (ddaspit) [reply lit_nlp/client/modules/attention_module.ts:207]: I wasn't sure if this is broadly useful, so I put it here. I can move it to utils.ts, if you would like.

Comment (EmilyReif) [reply lit_nlp/client/modules/attention_module.ts:207]: Yes, that would be great, thanks-- token spacing comes up pretty frequently

**Review by EmilyReif — APPROVED** [review]

File: lit_nlp/client/modules/attention_module.ts (hunk 1 above)
Comment (ddaspit) [reply lit_nlp/client/modules/attention_module.ts]: Done

**Comment by ddaspit** [comment]
I needed to add `@material/mwc-icon-button-toggle` to `package.json` in order to get this build. I guess a recent change added a dependency on this package.

**Comment by EmilyReif** [comment]
LGTM. Thanks again for doing this.

**Pull request merged**